openedx/features/course_experience/tests/views/test_welcome_message.py

```diff
@@ -61,30 +77,33 @@ def tearDown(self):
         remove_course_updates(self.user, self.course)
         super(TestWelcomeMessageView, self).tearDown()
 
-    def test_welcome_message(self):
+    @ddt.data(welcome_message_url, latest_update_url)
+    def test_message_display(self, url_generator):
         create_course_update(self.course, self.user, 'First Update', date='January 1, 2000')
         create_course_update(self.course, self.user, 'Second Update', date='January 1, 2017')
         create_course_update(self.course, self.user, 'Retroactive Update', date='January 1, 2010')
-        response = self.client.get(welcome_message_url(self.course))
+        response = self.client.get(url_generator(self.course))
         self.assertEqual(response.status_code, 200)
         self.assertContains(response, 'Second Update')
         self.assertContains(response, 'Dismiss')
 
-    def test_replace_urls(self):
+    @ddt.data(welcome_message_url, latest_update_url)
+    def test_replace_urls(self, url_generator):
         img_url = 'img.png'
         create_course_update(self.course, self.user, "<img src='/static/{url}'>".format(url=img_url))
-        response = self.client.get(welcome_message_url(self.course))
+        response = self.client.get(url_generator(self.course))
         self.assertContains(response, "/asset-v1:{org}+{course}+{run}+type@asset+block/img.png".format(
             org=self.course.id.org,
             course=self.course.id.course,
             run=self.course.id.run
         ))
 
-    def test_empty_welcome_message(self):
-        response = self.client.get(welcome_message_url(self.course))
+    @ddt.data(welcome_message_url, latest_update_url)
+    def test_empty_message(self, url_generator):
+        response = self.client.get(url_generator(self.course))
         self.assertEqual(response.status_code, 204)
 
-    def test_dismiss_message(self):
+    def test_dismiss_welcome_message(self):
```

Comment:
The dismiss behavior exists entirely in JS, so the JS test should cover this behavior.